cassandra/metadata.py

```diff
@@ -2326,7 +2326,7 @@ def _build_table_metadata(self, row, col_rows=None, trigger_rows=None, index_row
             table_meta.options = self._build_table_options(row)
             flags = row.get('flags', set())
             if flags:
-                compact_static = False
+                compact_static = 'dense' not in flags and 'super' not in flags and 'compound' not in flags
```

Comment:
I don't know much about compact storage... have you validated with the C* spec that this is correct?

Reply:
If we follow the logic here,
old logic:
if we have flags, `compact_static` is set to `False`, otherwise `compact_static` is set to `True`. This is because `dense` and `super` directly affect the compact storage, and `compound` refers to CQL.

However, there's also the `counter` flag. When `counter` is set, `compact_static` can still be `True`, but we were incorrectly setting it `False` here. Instead of setting `compact_static` to `False` if there are any flags, we should only be setting it if we see one of the 3 flags that make it `False`.

Reply:
Ok. 

nit: move `is_dense = 'dense' in flags` some line before and use it in the compact_static and table_meta.is_compact_storage assignments